### PR TITLE
[FLINK-31351][sql-gateway] Don't stop the stuck thread by force

### PR DESCRIPTION
**Pull request by fsk119**
## What is the purpose of the change

*We should not stop the thread by force because it may cause an inconsistent state. So we just notify the users it has the possibility to cause resource leak.*

### CODE REVIEW

File: flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/operation/OperationManager.java [hunk 1]
```diff
@@ -183,12 +190,15 @@ public ResultSet fetchResults(
     /** Closes the {@link OperationManager} and all operations. */
     public void close() {
         stateLock.writeLock().lock();
+        Exception closeException = null;
         try {
             isRunning = false;
-            for (Operation operation : submittedOperations.values()) {
-                operation.close();
-            }
+            List<Operation> copiedOperations = new ArrayList<>(submittedOperations.values());
```
Comment (lincoln-lil) [inline flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/operation/OperationManager.java:196]: we can remove this copy and move the 'clear()' to finally clause

File: flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/operation/OperationManager.java [hunk 2]
```diff
@@ -30,26 +30,33 @@
...
 import java.util.concurrent.FutureTask;
 import java.util.concurrent.Semaphore;
+import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicReference;
 import java.util.concurrent.locks.ReadWriteLock;
 import java.util.concurrent.locks.ReentrantReadWriteLock;
 import java.util.function.Function;
 import java.util.function.Supplier;
 
+import static org.apache.flink.shaded.guava30.com.google.common.util.concurrent.Uninterruptibles.sleepUninterruptibly;
```
Comment (lincoln-lil) [inline flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/operation/OperationManager.java:58]: nit: just use normal import?

**Review by lincoln-lil — COMMENTED** [review]
@fsk119 LGTM! Only two minor comments.

**Review by lincoln-lil — APPROVED** [review]

File: flink-table/flink-sql-gateway/src/test/java/org/apache/flink/table/gateway/service/operation/OperationManagerTest.java [hunk 3]
```diff
@@ -64,8 +66,8 @@ public class OperationManagerTest {
             new ExecutorThreadFactory(
                     "SqlGatewayService Test Pool", IgnoreExceptionHandler.INSTANCE);
 
-    @BeforeAll
-    public static void setUp() {
+    @BeforeEach
+    public void setUp() {
```
Comment (snuyanzin) [inline flink-table/flink-sql-gateway/src/test/java/org/apache/flink/table/gateway/service/operation/OperationManagerTest.java:70]: nit: modifier could be package private
```suggestion
    void setUp() {
```

File: flink-table/flink-sql-gateway/src/test/java/org/apache/flink/table/gateway/service/operation/OperationManagerTest.java [hunk 4]
```diff
@@ -129,29 +135,54 @@ public void testCancelOperation() throws Exception {
     }
 
     @Test
-    public void testCancelOperationByForce() throws Exception {
-        AtomicReference<Throwable> exception = new AtomicReference<>(null);
+    public void testCancelUninterruptedOperation() throws Exception {
```
Comment (snuyanzin) [inline flink-table/flink-sql-gateway/src/test/java/org/apache/flink/table/gateway/service/operation/OperationManagerTest.java:138]: nit: modifier could be package private
```suggestion
    void testCancelUninterruptedOperation() throws Exception {
```

**Comment by snuyanzin** [comment]
lgtm, just a couple of minor comments

**Comment by fsk119** [comment]
Thanks a lot for @snuyanzin review. Because all comments are in about the `public` modifier, so I fix the comments locally.

Merging...

**Pull request closed without merging**